test/WebJobs.Script.Tests.Integration/ApplicationInsights/ApplicationInsightsEndToEndTestsBase.cs

```diff
@@ -4,85 +4,178 @@
 using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Http.Headers;
 using System.Reflection;
 using System.Threading.Tasks;
 using Microsoft.ApplicationInsights.Channel;
 using Microsoft.ApplicationInsights.DataContracts;
+using Microsoft.Azure.WebJobs.Host;
 using Microsoft.Azure.WebJobs.Logging;
-using Microsoft.Extensions.Logging;
-using Newtonsoft.Json;
 using Newtonsoft.Json.Linq;
 using Xunit;
 
 namespace Microsoft.Azure.WebJobs.Script.Tests.ApplicationInsights
 {
-    public abstract class ApplicationInsightsEndToEndTestsBase<TTestFixture> :
-        EndToEndTestsBase<TTestFixture> where TTestFixture : ApplicationInsightsTestFixture, new()
+    public abstract class ApplicationInsightsEndToEndTestsBase<TTestFixture>
+        : IClassFixture<TTestFixture> where TTestFixture : ApplicationInsightsTestFixture, new()
     {
-        public ApplicationInsightsEndToEndTestsBase(TTestFixture fixture) : base(fixture)
+        private ApplicationInsightsTestFixture _fixture;
+
+        public ApplicationInsightsEndToEndTestsBase(ApplicationInsightsTestFixture fixture)
         {
+            TestHelpers.ClearFunctionLogs("Scenarios");
+            TestHelpers.ClearFunctionLogs("Http-Scenarios");
+
+            _fixture = fixture;
         }
 
-        protected async Task ApplicationInsights_SucceedsTest()
+        [Fact]
+        public async Task Validate_Manual()
         {
             string functionName = "Scenarios";
-            TestHelpers.ClearFunctionLogs(functionName);
-
-            string functionTrace = $"Function trace: {Guid.NewGuid().ToString()}";
             int invocationCount = 5;
 
+            List<string> functionTraces = new List<string>();
+
+            // We want to invoke this multiple times specifically to make sure Node invocationIds
+            // are correctly being set. Invoke them all first, then come back and validate all.
             for (int i = 0; i < invocationCount; i++)
             {
-                ScenarioInput input = new ScenarioInput
+                HttpRequestMessage request = new HttpRequestMessage
```

Comment:
why are we using HttpRequestMessage? Is this because app insights only runs in a windows context?

Reply:
These are coming straight from the current tests. Addressing issues with the last batch of commits that came through. App Insights, (what we're using here) is not tied to Windows. 
This will continue to work with the current HTTP Trigger implementation. I'm tracking an the work to migrate this and other tests to the new model.